include/seqan3/argument_parser/validators.hpp

```diff
@@ -322,14 +323,14 @@ class file_ext_validator // TODO AFTER INITIAL MERGE: Check if file exists, also
         extensions{v}
     {}
 
-    /*!\brief Tests whether cmp lies inside extensions.
-     * \param cmp The input value to check.
+    /*!\brief Tests whether path lies inside extensions.
+     * \param path The input value to check.
      * \throws parser_invalid_argument
      */
-    void operator()(std::string const & cmp) const
+    void operator()(filesystem::path const & path) const
     {
-        std::string ext{cmp.substr(cmp.find_last_of('.') + 1)};
-
+        std::string ext{path.extension().string()};
+        ext = ext.substr(std::min(1, static_cast<int>(ext.size()))); // drop '.' if extension is non-empty
```

Comment:
I am not really hooked on filesystem::path::extension(), because it can return empty strings, too, which we never want, do we?

Getting the extension with views could look like this:
```cpp
auto v = path | view::reverse | view::take_until_or_throw(is_char<'.'>) | view::reverse;
```

We could even save this as `view::file_extension`...

Reply:
> which we never want, do we?

I see no problem since an empty extension will not match any valid extensions.

Your example would need an extra try catch block (we only want validation exceptions to be thrown on failure).

And it would also return an empty string on `.` as a file name.

Reply:
@h-2 Can I resolves this?